uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
```

Comment:
the name really doesn't inspire confidence. Either it does or it doesn't, let's not "attempt" anything. You usually don't find methods like `tryToDoXYZ()` and for good reason

Reply:
@AkA84 Unfortunately, I attempted to find a better name a couple of times for that func and I still cannot find a better name. Could you help me please?

Reply:
`performBalanceChangeCalculation`

Reply:
`perform` means `do`, `execute`... This is essentially the same as `calculateBalanceChange`

Reply:
I am now thinking perform actually works. Thank you.